### PR TITLE
[MRG]: Fweight and aweight support for np.var

### LINKED ISSUE #5164: ENH: Weight support for np.var
Is there any reason why there is no `weights` keyword for `np.var` similar to `np.average`? We sometimes need to custom code for this in sklearn sometimes, for example, [Here](https://github.com/scikit-learn/scikit-learn/pull/3474/files#diff-383546d17769864242e4668d2338b449R136)

If this is agreed upon, I can submit a Pull Request.

### PR DESCRIPTION
**Pull request by MechCoder**
Support for frequency weights and aweights in np.var. I've kept it coherent with `np.cov` and used the same consistency when ddof is not equal to zero or one.
Fixes https://github.com/numpy/numpy/issues/5164

### CODE REVIEW

**Comment by MechCoder** [comment]
cc: @charris 
I finally got some time to work on this. This might not be that important especially with the GSoC going on, but it might be good to have.

**Comment by MechCoder** [comment]
bump?

File: numpy/lib/function_base.py [hunk 1]
```diff
@@ -44,6 +44,45 @@
     ]
 
 
+def _product_fweights_aweights(fweights, aweights, shape):
+    """Get the product and validate fweights and aweights.
+    Used in weighted variance and cobariance calculation.
```
Comment (argriffing) [inline numpy/lib/function_base.py]: typo `cobariance`

**Comment by OmerJog** [comment]
What is the status of this PR?

**Comment by rgommers** [comment]
> What is the status of this PR?

hmm, this never got followed up on.

The two types of weights are weird imho. Also in `np.cov` they're weird, explanation isn't clear and no example. It doesn't look too good. I'd be in favor of closing this, or otherwise just using normal weights.

**Comment by charris** [comment]
@rgommers IIRC, there was quite a long discussion about the two weights.

**Comment by rgommers** [comment]
maybe we can find it and document it properly then? it doesn't make too much sense right now imho .....

**Comment by charris** [comment]
I can't find it with a quick search, but there was [this](https://blogs.sas.com/content/iml/2017/10/02/weight-variables-in-statistics-sas.html), that tickles my memory.

**Comment by tpoole** [comment]
The previous discussion can be found here: https://github.com/numpy/numpy/pull/4960

**Comment by OmerJog** [comment]
So was this abandoned since there is something missing in the PR or it just can be re-based and merged?

**Comment by seberg** [comment]
Considering the age and the fact that it requires quite a bit of rebasing:  I am going to close the PR.  I will add a note to gh-8581 that the PR is stale but could be a basis for bringing this up again.

Thanks for opening the initial PR!

**Pull request closed without merging**